internal/transport/mqtt.go

```diff
@@ -190,7 +195,13 @@ func (t *MQTT) Tx(
 	opts := t.client.OptionsReader()
 	topic := fmt.Sprintf("%v/%v/%v/out", config.DefaultConfig.PathPrefix, opts.ClientID(), addr)
 
-	if token := t.client.Publish(topic, 1, false, data); token.Wait() && token.Error() != nil {
+	token := t.client.Publish(topic, 1, false, data)
+	if !token.WaitTimeout(config.DefaultConfig.MQTTPublishTimeout) {
+		return TxResponseErr, nil, nil, fmt.Errorf(
+			"cannot publish message: connection timeout reached",
+		)
```

Comment:
Similar comment here. When you can show some value causing error state, then show it. I would change the code to something like this:

```go
return TxResponseErr, nil, nil, fmt.Errorf(
    "cannot publish message: connection timeout %v reached",
    config.DefaultConfig.MQTTPublishTimeout,
)
```

Reply:
Good point.